receiver/sqlserverreceiver/scraper.go

```diff
@@ -299,3 +326,243 @@ func (s *sqlServerScraperHelper) recordDatabaseStatusMetrics(ctx context.Context
+
+		// handling `total_logical_writes`
+		logicalWritesVal, err := strconv.ParseInt(row[logicalWrites], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		}
+		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, logicalWrites, logicalWritesVal); cached {
+			record.Attributes().PutInt(dbPrefix+logicalWrites, diff)
+		}
+
+		// handling `physical_reads`
+		physicalReadsVal, err := strconv.ParseInt(row[physicalReads], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		}
+		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, physicalReads, physicalReadsVal); cached {
+			record.Attributes().PutInt(dbPrefix+physicalReads, diff)
+		}
+
+		// handling `execution_count`
+		totalExecutionCount, err := strconv.ParseInt(row[executionCount], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		} else {
+			if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, executionCount, totalExecutionCount); cached {
+				record.Attributes().PutInt(dbPrefix+executionCount, diff)
+			}
+		}
+
+		// handle `total_worker_time`, storing milliseconds
+		workerTime, err := strconv.ParseInt(row[totalWorkerTime], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		} else {
+			if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, totalWorkerTime, workerTime/1000); cached {
+				record.Attributes().PutInt(dbPrefix+totalWorkerTime, diff)
+			}
+		}
+
+		// handle `total_grant_kb`
+		memoryGranted, err := strconv.ParseInt(row[totalGrant], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		} else {
+			if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, totalGrant, memoryGranted); cached {
+				record.Attributes().PutInt(dbPrefix+totalGrant, diff)
+			}
+		}
+
+		// handling `query_text`
+		obfuscatedSQL, err := obfuscateSQL(row[queryText])
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		}
+		record.Attributes().PutStr(dbPrefix+queryText, obfuscatedSQL)
+
+		// handling `query_plan`
+		obfuscatedQueryPlan, err := obfuscateXMLPlan(row[queryPlan])
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		}
+		record.Attributes().PutStr(dbPrefix+queryPlan, obfuscatedQueryPlan)
+	}
+
+	return logs, errors.Join(errs...)
+}
+
+// cacheAndDiff store row(in int) with query hash and query plan hash variables
+// (1) returns true if the key is cached before
+// (2) returns positive value if the value is larger than the cached value
+func (s *sqlServerScraperHelper) cacheAndDiff(queryHash string, queryPlanHash string, column string, val int64) (bool, int64) {
+	if s.cache == nil {
+		s.logger.Error("LRU cache is not successfully initialized, skipping caching and diffing")
+		return false, 0
+	}
```

Comment:
If this is an expected path that can be enabled by the user, because this is going to be very noisy.

Reply:
removed as now s.cache could never be nil